staging_vespalib/src/tests/clock/clock_benchmark.cpp

```diff
@@ -0,0 +1,171 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include <vespa/vespalib/util/clock.h>
+#include <cassert>
+#include <vector>
+#include <atomic>
+#include <cstring>
+
+using vespalib::Clock;
+using fastos::TimeStamp;
+
+struct UpdateClock {
+    virtual ~UpdateClock() {}
+    virtual void update() = 0;
+};
+
+struct NSValue : public UpdateClock {
+    void update() override { _value = std::chrono::steady_clock::now().time_since_epoch().count(); }
+    int64_t _value;
+};
+
+struct NSVolatile : public UpdateClock {
+    void update() override { _value = std::chrono::steady_clock::now().time_since_epoch().count(); }
+    volatile int64_t _value;
+};
```

Comment:
We should never want to use `volatile` over `atomic`, so I'm not sure if we need this? It's not well defined for concurrent access, and it's theoretically highly compiler dependent what we actually get out.

Reply:
I just wanted it here top compare performance and generated code.